WcaOnRails/app/models/competition.rb

```diff
@@ -156,6 +156,59 @@ def user_cannot_demote_themself
     end
   end
 
+  attr_accessor :receive_registration_emails
+  def receive_registration_emails=(r)
+    @receive_registration_emails = ActiveRecord::Type::Boolean.new.type_cast_from_database(r)
+  end
+
+  def receiving_registration_emails?(user_id)
+    competition_delegate = competition_delegates.find_by_delegate_id(user_id)
+    if competition_delegate && competition_delegate.receive_registration_emails
+      return true
+    end
+    competition_organizer = competition_organizers.find_by_organizer_id(user_id)
+    if competition_organizer && competition_organizer.receive_registration_emails
+      return true
+    end
+
+    return false
```

Comment:
I feel like this method could be implemented using one SQL query. Not sure if this would be more readable, though.